python-backend/app/api/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -0,0 +1,94 @@
+    permit_amendment_status_code = db.Column(
+        db.String(2), db.ForeignKey('permit_amendment_status_code.permit_amendment_status_code'))
+    permit_amendment_type_code = db.Column(
+        db.String(2), db.ForeignKey('permit_amendment_type_code.permit_amendment_type_code'))
+
+    def json(self):
+        return {
+            'permit_amendment_id':
+            self.permit_amendment_id,
+            'permit_amendment_guid':
+            str(self.permit_amendment_guid),
+            'permit_id':
+            self.permit_id,
+            'permit_amendment_status_code':
+            self.permit_amendment_status_code,
+            'permit_amendment_type_code':
+            self.permit_amendment_type_code,
+            'received_date':
+            self.received_date.isoformat() if self.received_date else None,
+            'issue_date':
+            self.issue_date.isoformat() if self.issue_date else None,
+            'authorization_end_date':
+            self.authorization_end_date.isoformat() if self.authorization_end_date else None
+        }
+
+    @classmethod
+    def find_by_permit_amendment_id(cls, _id):
+        return cls.query.filter_by(permit_amendment_id=_id).first()
+
+    @classmethod
+    def find_by_permit_id(cls, _id):
+        return cls.query.filter_by(permit_id=_id)
+
+    @validates('permit_amendment_status_code')
+    def validate_status_code(self, key, permit_amendment_status_code):
+        if not permit_amendment_status_code:
+            raise AssertionError('Permit amendment status code is not provided.')
+        return permit_amendment_status_code
+
+    @validates('permit_amendment_type_code')
+    def validate_type_code(self, key, permit_amendment_type_code):
+        if not permit_amendment_type_code:
+            raise AssertionError('Permit amendment type code is not provided.')
+        return permit_amendment_type_code
+
+    @validates('permit_id')
+    def validate_permit_id(self, key, permit_id):
+        if not permit_id:
+            raise AssertionError('Permit id is not provided.')
+        return permit_id
+
+    @validates('received_date')
+    def validate_received_date(self, key, received_date):
+        if received_date.isoformat() == '9999-12-31':
+            raise AssertionError(
+                'Permit amendment received date should be set to null if not known.')
+        if received_date > datetime.today():
+            raise AssertionError('Permit amendment received date cannot be set to the future.')
+        return received_date
+
+    @validates('issue_date')
+    def validate_issue_date(self, key, issue_date):
+        if issue_date.isoformat() == '9999-12-31':
+            raise AssertionError('Permit amendment issue date should be set to null if not known.')
+        if issue_date > datetime.today():
+            raise AssertionError('Permit amendment issue date cannot be set to the future.')
+        return issue_date
```

Comment:
What is the intent of these validators? is it to transform/coalesce '9999-12-31' to None/null? Because this is currently going to throw an attribute error in it's current form.

Reply:
Not to transform 9999 to null, but issue date does not expect to follow the 9999 pattern for any use. It wouldn't make sense to set 9999 unless this app actually is being used in the year 9999.